crates/edr_napi/src/config.rs

```diff
@@ -362,16 +367,23 @@ impl ObservabilityConfig {
 
                     let on_collected_coverage_fn: Box<dyn SyncOnCollectedCoverageCallback> =
                         Box::new(move |hits| {
+                            let runtime = runtime.clone();
+
                             let (sender, receiver) = std::sync::mpsc::channel();
 
                             let status = on_collected_coverage_callback
-                                .call_with_return_value(hits, ThreadsafeFunctionCallMode::Blocking, move |()| {
-                                    sender.send(()).map_err(|_error| {
-                                        napi::Error::new(
-                                            napi::Status::GenericFailure,
-                                            "Failed to send result from on_collected_coverage_callback",
-                                        )
-                                    })
+                                .call_with_return_value(hits, ThreadsafeFunctionCallMode::Blocking, move |result: Promise<()>| {
+                                    // We spawn a background task to handle the async callback
+                                    runtime.spawn(async move {
+                                        let () = result.await?;
```

Comment:
If the promise rejects, `sender` is dropped and `receiver.recv()` below will error and panic, which I don't think we want. We should probably ignore the promise result:
```suggestion
                                        let _ = result.await;
```

Reply:
Good catch!